### PR TITLE
[Event Hubs Client] Processor Options Fix

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to correct an issue where the load balancing interval specified in the processing options was not respected and to add the intended timing options to the `EventProcessorClientOptions`, which, despite being part of the approved design, were overlooked and not implemented as part of the load balancing strategy changes.

**Note:** These changes have been reviewed and approved by the architecture board and the .NET language architect. Marking with the corresponding tag for tracking purposes.

# Last Upstream Rebase

Tuesday, October 6, 10:42am (EST)

# References

- [Load Balancing Strategy Design](https://gist.github.com/srnagar/366cb66279406db738f8eff82fbfb851#gistcomment-3321729)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs [hunk 1]
```diff
@@ -207,6 +213,44 @@ public int PrefetchCount
...
+        ///   The desired amount of time to allow between load balancing verification attempts.
+        /// </summary>
+        ///
+        /// <remarks>
+        ///   Because load balancing holds less priority than processing events, this interval
+        ///   should be considered the minimum time that will elapse between verification attempts; operations
+        ///   with higher priority may cause a minor delay longer than this interval for load balancing.
+        /// </remarks>
+        ///
+        public TimeSpan LoadBalancingUpdateInterval
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs:226]: Should the default value be mentioned?

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs [hunk 2]
```diff
@@ -207,6 +213,44 @@ public int PrefetchCount
...
+            set
+            {
+                Argument.AssertNotNegative(value, nameof(LoadBalancingUpdateInterval));
+                _loadBalancingUpdateInterval = value;
+            }
+        }
+
+        /// <summary>
+        ///   The desired amount of time to consider a partition owned by a specific event processor
+        ///   instance before the ownership is considered stale and the partition eligible to be requested
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs:239]:
```suggestion
        ///   instance before the ownership is considered stale and the partition becomes eligible to be requested
```

File: sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.Infrastructure.cs [hunk 3]
```diff
@@ -242,5 +242,30 @@ public async Task ProcessorLoadBalancerUsesTheExpectedStorageManager()
...
                 .Verify<Task<IEnumerable<EventProcessorPartitionOwnership>>>("ClaimOwnershipAsync", Times.Once(), ItExpr.IsAny<IEnumerable<EventProcessorPartitionOwnership>>(), ItExpr.IsAny<CancellationToken>());
         }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        public void ProcessorLoadBalanceIsConfiguredUsingOptions()
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.Infrastructure.cs:252]: Typo has been fixed; will be pushed once the initial test passes complete.

**Review by christothes — APPROVED** [review]
LHTM

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs:226]: Good thought.    Added in.

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClientOptions.cs:239]: Applied manually to avoid confusing my rebase.

**Pull request merged**